### ISSUE TITLE
YAML highlight

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#597]
(Original issue 890 created by piuccio on 2013-07-25T15:44:33.467617+00:00)

Hello,
I've tried to style YAML files but I think the parser should be improved

The following code
````
resources: some text
````

generates

````
<span class="l-Scalar-Plain">resources</span>
<span class="p-Indicator">resources</span>
<span class="l-Scalar-Plain">some text</span>
````

It should be something like
````
<span class="nt">resources</span>
<span class="p">resources</span>
<span class="sx">some text</span>
````

`resources` is a tag and `some text` is unquoted string

**Comment by Anteru** [comment]
(Original issue 890 last updated on 2018-12-13T18:59:48.300516+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by piuccio on 2013-07-25T15:44:59.194081+00:00)

This project might help https://github.com/textmate/yaml.tmbundle

**Comment by Anteru** [comment]
(Original comment by pathawks on 2014-11-17T17:46:00.184669+00:00)

Because YAMLLexer uses it's own tokens, existing styles do not work when using Pygments with YAML.

**Comment by Anteru** [comment]
(Original comment by thockin on 2015-07-19T04:27:37.972172+00:00)

Is there no plan to make this work?

Is there at least a list of all the CSS ids that yaml code will use, so we can manually style it?

**Comment by Anteru** [comment]
(Original comment by whitelynx on 2016-08-23T15:12:37.420128+00:00)

It seems to me this should actually be marked as a bug instead of an enhancement; as it is, using `pygmentize -l yaml` on the command line seems to highlight nothing but comments, which makes it next to useless. It seems that we should be able to use a dictionary lookup to map from YAMLLexer's tokens to the standard Pygments tokens.

**Comment by Anteru** [comment]
(Original comment by jessetan on 2017-11-08T13:09:33.476490+00:00)

Mapping YAMLLexer tokens to Pygments tokens would also require changing the lexer; current YAML lexer marks almost everything that is not a comment or punctuation in the `example.yaml` as `Literal.Scalar.Plain`, which means that everything will be syntax highlighted in the same color.
Other syntax highlighters have lexers that seem to better at differentiating keys and values.

**Comment by Anteru** [comment]
(Original comment by felixfontein on 2018-05-19T14:22:39.422349+00:00)

I tried to improve this in https://bitbucket.org/birkenfeld/pygments-main/pull-requests/762/

**Comment by Anteru** [comment]
(Original comment by Anteru on 2018-12-13T18:59:42.305171+00:00)

Merged, thanks!

**Comment by Anteru** [comment]
(Original comment by Anteru on 2018-12-13T18:59:48.331689+00:00)

Resolved in 98bdd9a1ebd2
